src/drivers/interrupt/cortexm_irq_handle.S

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file
+ * @brief
+ *
+ * @author  Alex Kalmuk
+ * @date    27.03.2017
+ */
+
+.text
+.thumb
+.syntax unified
+.global interrupt_handle_enter
+interrupt_handle_enter:
+
+    mov    r0, sp
+    sub    sp, #58
+    str    r0, [sp, #56]
+    stmia  sp, {r0 - r12}
+    str    lr, [sp, #52]
+
+    mov    r0, sp
+    bl     interrupt_handle
+
+.text
+.thumb
+.syntax unified
+.global __irq_trampoline
+__irq_trampoline:
+
+    cpsid  i
+    # r0 is a pointer to struct irq_saved_state
+    ldr    sp,  [r0, #32]
+    ldr    r14, [r0, #36]
+    bx     r14
```

Comment:
This suppose to return from exception or to just call further? I can't understand the rest without understanding this.

(!)Anyway, sp is going to be restored at the state of exception (start of interrupt_handle_enter). Exceptional state, context *regs and irq_saved_state `state` could be overwritten